File: src/main/java/Ex03_Classes.java
public class Ex03_Classes {
    public static void main(String[] args) {
        Apple apple = new Apple("red");
        Avocado avocado = new Avocado();
        System.out.printf("The %s %s have %s sugar%n", apple.color(), apple.name(), apple.sugar());
        System.out.printf("The %s have %s sugar%n", avocado.name(), avocado.sugar());
    }
    sealed interface Eatable permits Apple, Avocado {}

    record Apple(String color) implements Eatable {

        public String name() { return "Apple"; }
        public int sugar() { return 50; }
    }

    record Avocado() implements Eatable {

        public String name() { return "Avocado"; }
        public int sugar() { return 20; }
    }
}

File: src/main/kotlin/Ex03_Classes.kt

fun main() {
  val banana = Banana("The Yellow Banana have 40 sugar")
  val apple = Apple("red")
  val avocado = Avocado
  println("The %s %s have %s sugar".format(apple.color, apple.name, apple.sugar))
  println("The %s have %s sugar".format(avocado.name, avocado.sugar))
  println(orange.toString())
  println(banana.example)
}

sealed interface Eatable

data class Apple( // Data class are final
  val color: String
) : Eatable {
  val name: String get() = "Apple"
  val sugar: Int get() = 50
}

object Avocado : Eatable {
  val name: String get() = "Avocado"
  val sugar: Int get() = 20
}

val orange = object {
  val name: String = "Orange"
  val sugar: Int = 60

  override fun toString(): String = "The $name have $sugar sugar"
}

@JvmInline
value class Banana(val example: String)